src/extras/amd.js

```diff
@@ -155,6 +155,9 @@
   function addToRegisterRegistry(name, define) {
     if (!firstNamedDefine) {
       firstNamedDefine = define;
+      setTimeout(function () {
+        firstNamedDefine = null;
+      });
```

Comment:
If we're clearing these here, do we still need the clearing in System.import and System.register?

Reply:
I think the System.import clearing is unnecessary with this new change. But I think the getRegister clearing might still be necessary.

The original reason I removed the setTimeout was due to the bug described in https://github.com/systemjs/systemjs/issues/2111. If you look at the browser console in [this code sandbox](https://codesandbox.io/s/serverless-fast-rltnf), you can see that sometimes multiple calls to getRegister() occur in the same tick of the event loop.

I was unable to create a test case that ensures multiple calls to getRegister() in same tick of event loop, so our tests don't cover this situation.

So the setTimeout doesn't fully solve it, but also the getRegister/import clearing doesn't fully solve it either. To be honest, I don't know if all of them together will fully solve it.

Reply:
> I think the System.import clearing is unnecessary with this new change. But I think the getRegister clearing might still be necessary.

Can we remove the System.import paths then?

As far as I'm aware, the immediate setTimeout will add to the task queue which will be flushed as soon as the getRegister function exists, so that there is no way short of nested getRegister that this can happen.

Just better to have focused code paths instead of trying a blast all approach, that is all.

Reply:
👍 updated. I verified that the tests I wrote in #2116 still are passing, which were the cases where the `import` hook was previously needed. So I think we shouldn't see any more problems with this.